### PR TITLE
fix(cruft): change GH update token

### LINKED ISSUE #11: fix(ci): github token for update
[refer](https://github.com/elixir-cloud-aai/tus-storagehandler/pull/10#issuecomment-2270618221)

### PR DESCRIPTION
**Pull request by JaeAeich**
#### Description

- Fixes #11 

#### Checklist

- [ ] My code follows the [contributing guidelines][contributing-guidelines] of this
    project, including, in particular, with regard to any style guidelines
- [ ] The title of my PR complies with the [Conventional Commits
    specification][conv-commits]; in particular, it clearly indicates
    that a change is a breaking change
- [ ] I acknowledge that all my commits will be squashed into a single commit,
    using the PR title as the commit message
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code in hard-to-understand areas
- [ ] I have updated the user-facing documentation to describe any new or
    changed behavior
- [ ] I have added type annotations for all function/class/method interfaces
    or updated existing ones (only for Python, TypeScript, etc.)
- [ ] I have provided appropriate documentation ([Google-style Python
    docstrings][py-doc-google]) for all packages/modules/functions/classes/
    methods or updated existing ones
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature
    works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have not reduced the existing code coverage

#### Comments

[contributing-guidelines]: https://elixir-cloud-aai.github.io/guides/guide-contributor/workflow/
[conv-commits]: https://www.conventionalcommits.org/en
[py-doc-google]: https://google.github.io/styleguide/pyguide.html

## Summary by Sourcery

Fix the GitHub token name in the update workflow and README. Refactor test fixtures to standard test functions in integration and unit tests.

Bug Fixes:
- Fix the GitHub token name in the update workflow to use the correct token.

Enhancements:
- Refactor test fixtures to standard test functions in integration and unit tests.

Documentation:
- Update README to reflect the correct GitHub token name.

### CODE REVIEW

**Comment by JaeAeich** [comment]
@uniqueg I am not sure about this but changed the token, for if we need we can merge it first.

AAH I see tus_handlers PR has already been merged. Leave it!! Then we can take care of this later. Close the PR please if this isn;t what you meant.

**Comment by JaeAeich** [comment]
how about we make changes to the repo this week, and you can just create a copy of the token with the GITHUB_TOKEN and add it to the org, if we get a PR on moday, all good and we can go ahead with this or else checkout whats wrong.

File: README.md [hunk 1]
```diff
@@ -166,7 +166,7 @@ settings:
     > repository that you want to set up (you may need to "Resync") and hit the
     > "Configure" button. The token should show up in the "Coverage" tab.
 
-- `AUTO_UPDATE_GITHUB_TOKEN`: GitHub token with permissions to write to the
+- `GITHUB_TOKEN`: GitHub token with permissions to write to the
     repository.
```
Comment (uniqueg) [inline README.md:170]: Please mention that this is auto-generated. We don't want anyone to set this manually and risk overriding the auto-generated one.

Also, I would shorten the text from the note (it's not important when the need for granting the authority was introduced) and include it in the variable description, e.g., sth like this:

> `GITHUB_TOKEN`: This token is auto-generated by GitHub Actions for each job, so please do **NOT** set it manually. However, the token needs to be granted the authority to create pull requests. See [GitHub][gh-update-token] for details on how to set this permission.

File: {{ cookiecutter.project_name }}/tests/test_integration/test_init.py
Comment (uniqueg) [inline {{ cookiecutter.project_name }}/tests/test_integration/test_init.py:1]: Why do these changes show up in this PR? That is from another PR, no? Maybe merge the latest updates from `main` into this feature branch to get rid of these?

**Review by uniqueg — CHANGES_REQUESTED** [review]

**Comment by JaeAeich** [comment]
@uniqueg maybe merge #15 before this :)

**Review by uniqueg — APPROVED** [review]

**Comment by JaeAeich** [comment]
merging this first, will add one more commit for the typo to #15 and merge after this.

**Pull request merged**